dicer/mappers/mapper_helper.py

```diff
@@ -0,0 +1,106 @@
+    elif dim_type_obj is DimensionType.Attribute:
+        return TLDimensionType.Attribute
+
+
+def value_type_object_to_value_type(type_obj: ValueType) -> TLValueType:
+    if type_obj is ValueType.String:
+        return TLValueType.Categorical
+    elif type_obj in [ValueType.Double, ValueType.Int]:
+        return TLValueType.Numeric
+    elif type_obj is ValueType.Timestamp:
+        return TLValueType.Date
+    elif type_obj is ValueType.Map:
+        return TLValueType.Text #TODO ?
+
+
+def observed_value_type_to_value_type(obs_value_type: ObservedValueType) -> TLValueType:
+    if obs_value_type in [ObservedValueType.Categorical, ObservedValueType.CategoricalOption]:
+        return TLValueType.Categorical
+    elif obs_value_type is ObservedValueType.Numeric:
+        return TLValueType.Numeric
+    elif obs_value_type is ObservedValueType.Date:
+        return TLValueType.Date
+    else:
+        return TLValueType.Text #TODO ?
+
+
+def value_by_value_type(value, value_type: TLValueType) -> TLValue:
+    if value_type is TLValueType.Categorical:
+        return TLCategoricalValue(value)
+    elif value_type is TLValueType.Numeric:
+        return TLNumericalValue(value)
+    elif value_type is TLValueType.Date:
+        return TLDateValue(value)
+    elif value_type is TLValueType.Text:
+        return TLTextValue(value)
+
+
+def map_relation_type(relation_type: RelationType) -> TLRelationType:
+    return TLRelationType(
+        relation_type.label,
+        relation_type.description,
+        relation_type.symmetrical,
+        relation_type.biological
+    )
+
+
+def map_modifier(modifier_dimension: DimensionObject) -> TLModifier:
+    return TLModifier(
+        modifier_dimension.modifierCode,
+        modifier_dimension.name,
+        modifier_dimension.modifierCode,
+        value_type_object_to_value_type(modifier_dimension.valueType)
+    )
+
+
+def map_dimension(dimension: DimensionObject, modifiers: List[TLModifier]) -> TLDimension:
+    modifier = next(filter(lambda x: x.modifier_code == dimension.modifierCode,  modifiers), None)
+    return TLDimension(
+        dimension.name,
+        modifier,
+        dimension_type_object_to_dimension_type(dimension.dimensionType),
+        dimension.sortIndex
+    )
+
+
+def map_trial_visit(trial_visit: TrialVisitDimensionElement, study_id_to_study: Dict[str, TLStudy]) -> TLTrialVisit:
+    study = study_id_to_study.get(trial_visit.studyId)
+    return TLTrialVisit(
+        study,
+        trial_visit.relTimeLabel,
+        trial_visit.relTimeUnit,
+        trial_visit.relTime
+    )
+
+
+def map_study(study: Study) -> TLStudy:
+    return TLStudy(
+        study.studyId,
+        study.studyId
+    )
+
+
+def get_study_id_to_study_dict(study_dim_elements: List[Value], all_studies: List[Study]):
+    study_id_to_study: Dict[str, TLStudy] = {}
+    for study in all_studies:
+        if study.studyId in list(map(lambda x: StudyDimensionElement(**x).name, study_dim_elements)):
```

Comment:
The `list(map(lambda x: StudyDimensionElement(**x).name, study_dim_elements))` could be outside of the `for` loop. I guess that `set` would make more sense, because the only operation is testing membership. But I don't know the performance differences between `list` and `set` in Python.

Reply:
right! From what I just checked determining whether an object is in the set is faster than in the list. I updated this function.